router/core/graphql_prehandler.go

```diff
@@ -315,11 +317,29 @@ func (h *PreHandler) Handler(next http.Handler) http.Handler {
 			body:             body,
 		})
 		if err != nil {
-			requestContext.error = err
+			outputErr := err
+			var httpErr ApolloCompatibleHttpError
+			var reportErr ReportError
+			if h.executor.Resolver.Options().ApolloCompatibilityReplaceInvalidVariableError && errors.As(err, &httpErr) {
+				outputErr = &operationreport.ApolloCompatibilityError{
+					ExtensionCode: httpErr.ExtensionCode(),
+					Message:       httpErr.Error(),
+				}
+			} else if h.executor.Resolver.Options().ApolloCompatibilityReplaceUndefinedOperationFieldError && errors.As(err, &reportErr) {
```

Comment:
I don't understand how you identified an `ReplaceUndefinedOperationField` error here. Also what happens when both modes are enabled? Does it mean `ApolloCompatibilityReplaceInvalidVariableError` has always precedence?

Reply:
The opposite. If the operation itself is invalid validation of variables won't happen.